de.metas.acct.base/src/main/java-legacy/org/compiere/acct/DocLine_Inventory.java

```diff
@@ -55,7 +58,12 @@ public DocLine_Inventory(final I_M_InventoryLine inventoryLine, final Doc_Invent
 			qty = qtyCount.subtract(qtyBook);
 		}
 
-		setQty(Quantity.of(qty, getProductStockingUOM()), false);
+		//calculate the cost price considering qty in inventory UOM
+		if (inventoryLine.getCostPrice() != null && qty.signum() > 0) {
```

Comment:
NOTE: intentoryLine.getCostPrice() always returns not null.
Basically we never return null in case of BigDecimals.

Better check it as `InterfaceWrapperHelper.isNull(inventoryLine, I_M_InventoryLine.COLUMNNAME_CostPrice)`.

or even better, don't check it at all. Check just the quantity.
Because in case it's zero or null would be like before => i.e. ZERO cost price

Reply:
Yep, that's right; on it 👍